indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalInsertAction.java

```diff
@@ -257,11 +260,27 @@ public SegmentPublishResult perform(Task task, TaskActionToolbox toolbox)
           segment.getShardSpec() == null ? null : segment.getShardSpec().getType()
       );
       toolbox.getEmitter().emit(metricBuilder.build("segment/added/bytes", segment.getSize()));
+      // Emit the segment related metadata using the configured emitters
+      this.emitSegmentMetadata(segment, toolbox);
```

Comment:
can we add a comment that captures the limitation that @gianm pointed out, explaining that some segmentmetadata events may not be captured in the event of server crash?